### PR TITLE
Using constructors instead of factory methods in SFBatchingSyncUpTarget

### PR DESCRIPTION
**Pull request by wmathurin**
More in line with super classes.

### CODE REVIEW

File: libs/SmartSync/SmartSync/Classes/SFBatchingSyncUpTarget.h [hunk 1]
```diff
@@ -34,12 +34,15 @@ NS_SWIFT_NAME(BatchingSyncUpTarget)
  */
 @interface SFBatchingSyncUpTarget : SFSyncUpTarget <SFAdvancedSyncUpTarget>
 
-/** Factory methods
+/** Constructor
  */
+- (instancetype)initWithCreateFieldlist:(nullable NSArray *)createFieldlist
```
Comment (trooper2013) [inline libs/SmartSync/SmartSync/Classes/SFBatchingSyncUpTarget.h]: @wmathurin Could we  specify the type in the list  like below?
``` (instancetype)initWithCreateFieldlist:(nullable NSArray<NSString *> *)createFieldlist  updateFieldlist:(nullable NSArray<NSString *> *)updateFieldlist maxBatchSize:(nullable NSNumber *)maxBatchSize; ```

Comment (bhariharan) [reply libs/SmartSync/SmartSync/Classes/SFBatchingSyncUpTarget.h]: Does this need `NS_SWIFT_NAME` magic too?

Comment (wmathurin) [reply libs/SmartSync/SmartSync/Classes/SFBatchingSyncUpTarget.h]: It's on line 31.

Comment (wmathurin) [reply libs/SmartSync/SmartSync/Classes/SFBatchingSyncUpTarget.h]: Here is the generated swift header
```swift
open class BatchingSyncUpTarget : SyncUpTarget, AdvancedSyncUpTarget {

    
    /** Constructor
     */
    public init(createFieldlist: [Any]?, updateFieldlist: [Any]?, maxBatchSize: NSNumber?)

    
    
    /** Factory method
     */
    open class func build(dict: [AnyHashable : Any]?) -> Self
}
```

**Comment by wmathurin** [comment]
After last commit, here is the generated Swift 4.2 header for SFBatchingSyncUpTarget
```swift
open class BatchingSyncUpTarget : SyncUpTarget, AdvancedSyncUpTarget {

    public init(createFieldlist: [String]?, updateFieldlist: [String]?, maxBatchSize: NSNumber?)

    open class func build(dict: [AnyHashable : Any]?) -> Self

}
```

**Pull request merged**